tests/components/ccm15/test_climate.py

```diff
@@ -0,0 +1,110 @@
+"""Unit test for CCM15 coordinator component."""
+from unittest.mock import AsyncMock, patch
+
+from syrupy.assertion import SnapshotAssertion
+
+from homeassistant.components.ccm15.const import DOMAIN
+from homeassistant.components.climate import (
+    ATTR_FAN_MODE,
+    ATTR_HVAC_MODE,
+    ATTR_TEMPERATURE,
+    DOMAIN as CLIMATE_DOMAIN,
+    FAN_HIGH,
+    SERVICE_SET_FAN_MODE,
+    SERVICE_SET_HVAC_MODE,
+    SERVICE_SET_TEMPERATURE,
+    SERVICE_TURN_ON,
+    HVACMode,
+)
+from homeassistant.const import ATTR_ENTITY_ID, CONF_HOST, CONF_PORT, SERVICE_TURN_OFF
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+
+from tests.common import MockConfigEntry
+
+
+async def test_climate_state(
+    hass: HomeAssistant,
+    snapshot: SnapshotAssertion,
+    entity_registry: er.EntityRegistry,
+    ccm15_device: AsyncMock,
+) -> None:
+    """Test the coordinator."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        unique_id="1.1.1.1",
+        data={
+            CONF_HOST: "1.1.1.1",
+            CONF_PORT: 80,
+        },
+    )
+    entry.add_to_hass(hass)
+
+    assert await hass.config_entries.async_setup(entry.entry_id)
+    await hass.async_block_till_done()
+
+    assert entity_registry.async_get("climate.0") == snapshot
+    assert entity_registry.async_get("climate.1") == snapshot
+
+    assert hass.states.get("climate.0") == snapshot
+    assert hass.states.get("climate.1") == snapshot
```

Comment:
The only thing I don't like about the whole integration is that the name of the devices is "0" and "1", I mean if we don't get a name or anything back from the API, so be it. But can we at least try to add some name in front of it?

(Also when you are now changing the name of the entity, run `pytest ./tests/components/ccm15 --snapshot-update` as it updates the .ambr file)

Reply:
What is the .ambr file?

Reply:
Check your dms and I'll explain more :)

Reply:
Used the naming "Midea {Index}"